public/app/features/expressions/ExpressionDatasource.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  DataSourceApi,
+  DataQueryRequest,
+  DataQueryResponse,
+  DataSourceInstanceSettings,
+  DataSourcePluginMeta,
+} from '@grafana/ui';
+import { ExpressionQuery, GELQueryType } from './types';
+import { ExpressionQueryEditor } from './ExpressionQueryEditor';
+import { Observable, from } from 'rxjs';
+import { config } from '@grafana/runtime';
+import { getBackendSrv } from 'app/core/services/backend_srv';
+import { gelResponseToDataFrames } from './util';
+
+/**
+ * This is a singleton that is not actually instanciated
+ */
+export class ExpressionDatasourceApi extends DataSourceApi<ExpressionQuery> {
+  constructor(instanceSettings: DataSourceInstanceSettings) {
+    super(instanceSettings);
+  }
+
+  getCollapsedText(query: ExpressionQuery) {
+    return `Expression: ${query.type}`;
+  }
+
+  query(request: DataQueryRequest): Observable<DataQueryResponse> {
+    const { targets, intervalMs, maxDataPoints, range } = request;
+
+    const orgId = (window as any).grafanaBootData.user.orgId;
+    const queries = targets.map(q => {
+      if (q.datasource === ExpressionDatasourceID) {
+        return {
+          ...q,
+          datasourceId: this.id,
+          orgId,
+        };
+      }
+      const ds = config.datasources[q.datasource || config.defaultDatasource];
+      return {
+        ...q,
+        datasourceId: ds.id,
+        intervalMs,
+        maxDataPoints,
+        orgId,
+        // ?? alias: templateSrv.replace(q.alias || ''),
+      };
+    });
+    return from(
+      (getBackendSrv().datasourceRequest({
+        method: 'POST',
+        url: '/api/tsdb/query/v2',
+        data: {
+          from: range.from.valueOf().toString(),
+          to: range.to.valueOf().toString(),
+          queries: queries,
+        },
+      }) as Promise<any>).then(res => {
```

Comment:
@torkelo -- I changed this to use `datasourceRequest`, that is not exposed on the runtime getBackendSrv().

NOTE: I tried to change the method types to return Promise, but it broke a bunch of datasources so I added the cast here

Reply:
Doesn't seem to be working for me with GEL:

```
Panel data error: TypeError: Cannot convert undefined or null to object
    at Function.values (<anonymous>)
    at gelResponseToDataFrames (default~app.777340c294f1a9edf1c6.js:47686)
    at default~app.777340c294f1a9edf1c6.js:47435
    at processQueue (angular~app.777340c294f1a9edf1c6.js:17178)
    at angular~app.777340c294f1a9edf1c6.js:17226
    at Scope.$digest (angular~app.777340c294f1a9edf1c6.js:18361)
    at Scope.$apply (angular~app.777340c294f1a9edf1c6.js:18658)
    at angular~app.777340c294f1a9edf1c6.js:18961
    at completeOutstandingRequest (angular~app.777340c294f1a9edf1c6.js:6437)
    at angular~app.777340c294f1a9edf1c6.js:6716
```

I do see results from the network call in Chrome inspector:

```
{"results":{"":{"refId":"","series":null,"tables":null,"dataframes":["QVJST1cxAAC8AQAAEAAAAAAACgAOAAwACwAEAAoAAAAUAAAAAAAAAQMACgAMAAAACAAEAAoAAAAIAAAAWAAAAAIAAAAoAAAABAAAAND+//8IAAAADAAAAAEAAABBAAAABQAAAHJlZklkAAAA8P7//wgAAAAUAAAACAAAAEEtc2VyaWVzAAAAAAQAAABuYW1lAAAAAAIAAACcAAAABAAAAH7///8UAAAAZAAAAGQAAAAAAAMBZAAAAAIAAAAsAAAABAAAAEj///8IAAAAEAAAAAYAAABudW1iZXIAAAQAAAB0eXBlAAAAAGz///8IAAAAEAAAAAUAAAB2YWx1ZQAAAAQAAABuYW1lAAAAAAAAAABi////AAACAAUAAAB2YWx1ZQASABgAFAATABIADAAAAAgABAASAAAAFAAAAGwAAAB0AAAAAAAKAXQAAAACAAAANAAAAAQAAADc////CAAAABAAAAAEAAAAdGltZQAAAAAEAAAAdHlwZQAAAAAIAAwACAAEAAgAAAAIAAAAEAAAAAQAAAB0aW1lAAAAAAQAAABuYW1lAAAAAAAAAAAAAAYACAAGAAYAAAAAAAMABAAAAHRpbWUAAAAAvAAAABQAAAAAAAAADAAWABQAEwAMAAQADAAAALAAAAAAAAAAFAAAAAAAAAMDAAoAGAAMAAgABAAKAAAAFAAAAFgAAAALAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAWAAAAAAAAABYAAAAAAAAAAAAAAAAAAAAWAAAAAAAAABYAAAAAAAAAAAAAAACAAAACwAAAAAAAAAAAAAAAAAAAAsAAAAAAAAAAAAAAAAAAAAAAAAAAJhC+m1r0hUA8Inye2vSFQBI0eqJa9IVAKAY45dr0hUA+F/bpWvSFQBQp9Oza9IVAKjuy8Fr0hUAADbEz2vSFQBYfbzda9IVALDEtOtr0hUACAyt+WvSFQAAAAAAAABAAAAAAAAA8D8AAAAAAADwPwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAABAAAAAAAAA8D8AAAAAAADwPwAAAAAAAAAAAAAAAAAAAAAQAAAADAAUABIADAAIAAQADAAAABAAAAAsAAAAOAAAAAAAAwABAAAAyAEAAAAAAADAAAAAAAAAALAAAAAAAAAAAAAAAAAAAAAAAAoADAAAAAgABAAKAAAACAAAAFgAAAACAAAAKAAAAAQAAADQ/v//CAAAAAwAAAABAAAAQQAAAAUAAAByZWZJZAAAAPD+//8IAAAAFAAAAAgAAABBLXNlcmllcwAAAAAEAAAAbmFtZQAAAAACAAAAnAAAAAQAAAB+////FAAAAGQAAABkAAAAAAADAWQAAAACAAAALAAAAAQAAABI////CAAAABAAAAAGAAAAbnVtYmVyAAAEAAAAdHlwZQAAAABs////CAAAABAAAAAFAAAAdmFsdWUAAAAEAAAAbmFtZQAAAAAAAAAAYv///wAAAgAFAAAAdmFsdWUAEgAYABQAEwASAAwAAAAIAAQAEgAAABQAAABsAAAAdAAAAAAACgF0AAAAAgAAADQAAAAEAAAA3P///wgAAAAQAAAABAAAAHRpbWUAAAAABAAAAHR5cGUAAAAACAAMAAgABAAIAAAACAAAABAAAAAEAAAAdGltZQAAAAAEAAAAbmFtZQAAAAAAAAAAAAAGAAgABgAGAAAAAAADAAQAAAB0aW1lAAAAAOgBAABBUlJPVzE=","QVJST1cxAACsAQAAEAAAAAAACgAOAAwACwAEAAoAAAAUAAAAAAAAAQMACgAMAAAACAAEAAoAAAAIAAAAUAAAAAIAAAAoAAAABAAAAOD+//8IAAAADAAAAAEAAABCAAAABQAAAHJlZklkAAAAAP///wgAAAAMAAAAAAAAAAAAAAAEAAAAbmFtZQAAAAACAAAAlAAAAAQAAACG////FAAAAGAAAABgAAAAAAADAWAAAAACAAAALAAAAAQAAABQ////CAAAABAAAAAGAAAAbnVtYmVyAAAEAAAAdHlwZQAAAAB0////CAAAAAwAAAAAAAAAAAAAAAQAAABuYW1lAAAAAAAAAABm////AAACAAAAAAAAABIAGAAUABMAEgAMAAAACAAEABIAAAAUAAAAbAAAAHQAAAAAAAoBdAAAAAIAAAA0AAAABAAAANz///8IAAAAEAAAAAQAAAB0aW1lAAAAAAQAAAB0eXBlAAAAAAgADAAIAAQACAAAAAgAAAAQAAAABAAAAFRpbWUAAAAABAAAAG5hbWUAAAAAAAAAAAAABgAIAAYABgAAAAAAAwAEAAAAVGltZQAAAAC8AAAAFAAAAAAAAAAMABYAFAATAAwABAAMAAAAsAAAAAAAAAAUAAAAAAAAAwMACgAYAAwACAAEAAoAAAAUAAAAWAAAAAsAAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABYAAAAAAAAAFgAAAAAAAAAAAAAAAAAAABYAAAAAAAAAFgAAAAAAAAAAAAAAAIAAAALAAAAAAAAAAAAAAAAAAAACwAAAAAAAAAAAAAAAAAAAAAAAAAAmEL6bWvSFQDwifJ7a9IVAEjR6olr0hUAoBjjl2vSFQD4X9ula9IVAFCn07Nr0hUAqO7LwWvSFQAANsTPa9IVAFh9vN1r0hUAsMS062vSFQAIDK35a9IVAAAAAAAAEEAAAAAAAAAIQAAAAAAAAAhAAAAAAAAAAEAAAAAAAAAAQAAAAAAAABBAAAAAAAAAEEAAAAAAAAAIQAAAAAAAAAhAAAAAAAAAAEAAAAAAAAAAQBAAAAAMABQAEgAMAAgABAAMAAAAEAAAACwAAAA4AAAAAAADAAEAAAC4AQAAAAAAAMAAAAAAAAAAsAAAAAAAAAAAAAAAAAAAAAAACgAMAAAACAAEAAoAAAAIAAAAUAAAAAIAAAAoAAAABAAAAOD+//8IAAAADAAAAAEAAABCAAAABQAAAHJlZklkAAAAAP///wgAAAAMAAAAAAAAAAAAAAAEAAAAbmFtZQAAAAACAAAAlAAAAAQAAACG////FAAAAGAAAABgAAAAAAADAWAAAAACAAAALAAAAAQAAABQ////CAAAABAAAAAGAAAAbnVtYmVyAAAEAAAAdHlwZQAAAAB0////CAAAAAwAAAAAAAAAAAAAAAQAAABuYW1lAAAAAAAAAABm////AAACAAAAAAAAABIAGAAUABMAEgAMAAAACAAEABIAAAAUAAAAbAAAAHQAAAAAAAoBdAAAAAIAAAA0AAAABAAAANz///8IAAAAEAAAAAQAAAB0aW1lAAAAAAQAAAB0eXBlAAAAAAgADAAIAAQACAAAAAgAAAAQAAAABAAAAFRpbWUAAAAABAAAAG5hbWUAAAAAAAAAAAAABgAIAAYABgAAAAAAAwAEAAAAVGltZQAAAADYAQAAQVJST1cx"]}}}```

Reply:
Seems like the promise is being rejected?